cloudigrade/account/util.py

```diff
@@ -88,22 +121,71 @@ def create_new_machine_images(account, instances_data):
         )
         new_amis = list(seen_amis.difference(known_amis))
 
-        for new_ami in new_amis:
-            ami = AwsMachineImage(
-                account=account,
-                ec2_ami_id=new_ami,
-            )
-            ami.save()
-            if new_ami in windows_instances:
-                ami.tags.add(ImageTag.objects.filter(
-                    description='windows').first())
-                ami.status = ami.INSPECTED
-                ami.save()
+        for ami_id in new_amis:
+            ami, __ = save_machine_images(account, ami_id)
+            if ami_id in windows_instances:
+                tag_windows(ami)
 
         saved_amis.extend(new_amis)
     return saved_amis
 
 
+def save_machine_images(account, ami_id):
+    """
+    Save the image object.
+
+    Args:
+        account (AwsAccount): The account associated with the machine image
+        ami_id (str): The AWS AMI ID.
+
+    Returns:
+
+    """
+    ami, new = AwsMachineImage.objects.get_or_create(
+        account=account,
+        ec2_ami_id=ami_id,
+    )
+
+    return ami, new
+
+
+def tag_windows(ami):
+    """
+    Tags the provided image with the windows tag.
+
+    Args:
+        ami (AwsMachineImage): Object representing the image.
+
+    Returns:
+        AwsMachineImage: Updated object.
+
+    """
+    ami.tags.add(ImageTag.objects.filter(description='windows').first())
+    ami.status = ami.INSPECTED
+    ami.save()
+
+    return ami
+
+
+def start_image_inspection(arn, image_id, region):
+    """
+    Start image inspection of the provided image.
+
+    Args:
+        arn (str):  The AWS Resource Number for the account with the snapshot
+        image_id (str): The AWS ID for the machine image
+        region (str): The region the snapshot resides in
+
+    """
+    image = AwsMachineImage.objects.get(ec2_ami_id=image_id)
+    image.status = image.PREPARING
+    image.save()
+
+    from account.tasks import copy_ami_snapshot
```

Comment:
Why are you importing this locally instead of at the start of the module? Is there a potential circular import that I've overlooked? If so, throw a comment in here to explain so that doesn't accidentally get refactored out.